schemaregistry/schemaregistry.md

```diff
@@ -404,4 +496,195 @@ The returned payload is the schema document. Further attributes such as the
+
+A peer-to-peer replication topology copies schemas between a source and target
+registry directly. For a mutual synchronization of schemas between registries, a
+second replication path is established with reversed roles.
+
+A hub replication topology copies schemas from a source to the target through an
+intermediary hub. The hub acts as an intermediary store, but with the authority
+over the schemas still being with the producers. Hubs might also be used by
+services that cannot host their own service registry endpoints.
+
+A star replication topology uses hubs central hubs into which all sources
+contribute schemas and to which all targets subscribe. If the hub is a centrally
+governed registry, it might not accept inbound replications.
+
+### 4.3 Push replication
+
+In an imperative "push" replication model, a source might push any updates of its
+state to one of more configured targets by forwarding all changes to the
+respective schema registry API calls on the target.
+
+The pushing source might be configured to copy changes of the full registry, of
+a specific schema group, or just of a specific schema.
+
+### 4.4 Pull replication
+
+In an imperative "pull" replication model, a target will once or periodically
+traverse the source registry contents and apply differences between the source
+state and its own state. 
+
+The pulling target might be configured to copy changes of the full registry, of
+a specific schema group, or just of a specific schema.
+
+### 4.5. Event-driven replication
+
+Event-driven replication uses CloudEvents to notify targets of changes in the
+source such that the targets can [pull](#44-pull-replication) changes
+immediately and as they become available.
+
+Each registry SHOULD offer a [Subscription API](../subscriptions-api.md)
+endpoint, either directly or using some middleware, to allow replication targets
+to subscribe to change events.
+
+The subscription's scope MUST be root of the schema registry. A [prefix
+filter](../subscriptions-api.md#3231-basic-filter-dialect) can be used to to
+scope the subscription to a particular schema group or schema.
+
+The subscribing target registers a notification sink of its choosing.
+
+```HTTP
+POST https://source.example.com/subscriptions HTTP/1.1
+Content-Type: application/cloudevents-subscription+json
+
+{
+  "id" : "mysubscription",
+  "scope": "https://source.example.com",
+  "protocol" : "HTTPS",
+  "sink" : "https://target.example.com/notifications",
+  "filter" : {
+    "dialect": "basic",
+    "filter": {
+        "type": "exact",
+        "property": "type",
+        "value": "io.cloudevents.schemaregistry.update"
+    },{
+        "type": "prefix",
+        "property": "subject",
+        "value": "/schemagroups/mygroup"
+    }
+}
+```
+
+#### io.cloudevents.schemaregistry.update
+
+This event notifies the subscriber of an element of the registry having been
+created or updated. Because schema documents might be very large, and because
```

Comment:
Any reason to not have two different events, one for "created" and one for "updated" ? Worried about events arriving out of order?

Reply:
The semantics are "the registry has been updated, look at the subject to find out what"

Schema versions are append-only and immutable, so that is always "create" and that is the majority of the events that would be raised. Schema groups may be updated. I think "upsert" semantics are sufficient here. "create" vs. "update" really only complicates matters here, IMO.